src/main/java/com/diffblue/deeptestutils/Reflector.java

```diff
@@ -130,33 +135,31 @@ public static void setField(
   public static <T> Object getInstanceField(
       final Class<T> c,
       final Object o,
-      final String fieldName)
-      throws
-      NoSuchFieldException,
-      IllegalArgumentException,
-      IllegalAccessException {
+      final String fieldName) {
     if (c == null) {
-      throw new NoSuchFieldException();
+      throw new DeeptestUtilsException(fieldName + " is not a field");
     }
-    Field field = null;
-    for (Field f : c.getDeclaredFields()) {
-      if (f.getName().equals(fieldName)) {
-        field = f;
-        break;
+      Field field = null;
+      for (Field f : c.getDeclaredFields()) {
+        if (f.getName().equals(fieldName)) {
+          field = f;
+          break;
+        }
       }
-    }
-    if (field == null) {
-      return getInstanceField(c.getSuperclass(), o, fieldName);
-    } else {
-      Field property = field;
-      property.setAccessible(true);
-
-      try {
-        return property.get(o);
-      } catch (IllegalAccessException ex) {
-        throw new RuntimeException(ex); // Should never happen.
+      if (field == null) {
+        return getInstanceField(c.getSuperclass(), o, fieldName);
+      } else {
+        Field property = field;
+        property.setAccessible(true);
+
+        try {
+          return property.get(o);
+        } catch (IllegalArgumentException e) {
+          throw new DeeptestUtilsException(e.getMessage(), e.getCause());
+        } catch (IllegalAccessException e) { // Should never happen.
```

Comment:
how will this be treated then?

Reply:
It's treated the same as before, but rather than throwing a general `RuntimeException` it's now throwing a `DeeptestUtilsException`